### ISSUE TITLE
Same color can be posted twice

### ISSUE THREAD

**Issue by meodai** [meodai/colorparrot-twitter-bot#4]
https://github.com/meodai/colorparrot-twitter-bot/blob/master/src/utils/twitter/sendRandomImage.js#L23

```javascript
  if (await checkIfColorExistsInTwitts(color.name)) { // 1)
    return sendRandomImage(T); // here can be a problem in the future!
  } else {
    await addColorNameInPostedTwitts(color.name); // 2
  }
```
1) You are checking if the color was already posted
2) If it was you add it to the posted tweets

This will never prevent an image to be posted twice, since the color is never added to the `addColorNameInPostedTwitts` is only called if the color exists already, what does not make any sense.

It should:

1) create Map where the mapkey is the name of the color and the value the hex color value
2) subtract all key from the already posted colors from the previously created map. (Or even smarter don't even include them in the map you create in the first place)
3) Tweet a color and add it to the list of posted tweets.

By the way its `tweets` not `twitts`

PS: If a color was requested by a user, It should still be able to be posted by the randomiser, remove: https://github.com/meodai/colorparrot-twitter-bot/blob/006e5a3eb7eb186d1eb67719cd7ab442be3255a5/src/middlewares/responseWithImage.js#L45

**Comment by NikolayKrishchuk** [comment]
example how it works

function sleep(ms){
    return new Promise(resolve=>{
        setTimeout(() => {
			const values = [1,2]
			resolve(values[Math.floor(Math.random() * values.length)])
		},ms)
    })
}

async function main(){
	if (await sleep(1000) === 1){
		main()
	} else {
		console.log('done')
	}
	
}

main()

**Comment by NikolayKrishchuk** [comment]
I will remove recursive call

**Comment by NikolayKrishchuk** [comment]
fixed, check it

**Comment by meodai** [comment]
Hey @NikolayKrishchuk please refer the issues in the commit messages, so they would show up here: Simply write something like "fixes issue #4". Thanks for the fix
